src/run/def.rs

```diff
@@ -200,25 +200,31 @@ impl<'a, M: Mode> Net<'a, M> {
 #[derive(Debug, Default, Clone)]
 pub struct InterpretedDef {
   pub instr: Vec<Instruction>,
+  pub trgs: usize,
 }
 
 impl AsDef for InterpretedDef {
   unsafe fn call<M: Mode>(slf: *const Def<Self>, net: &mut Net<M>, trg: Port) {
-    let instructions = unsafe { &(*slf).data.instr };
+    let def = unsafe { &(*slf).data };
+    let instructions = &def.instr;
 
-    let mut trgs = unsafe { std::mem::transmute::<_, Trgs>(Trgs(&mut net.trgs[..])) };
+    if def.trgs >= net.trgs.len() {
+      net.trgs = Box::new_uninit_slice(def.trgs);
+    }
+
+    let mut trgs = Trgs(&mut net.trgs[..] as *mut _ as *mut _);
 
-    struct Trgs<'a>(&'a mut [Trg]);
+    struct Trgs(*mut Trg);
 
-    impl<'a> Trgs<'a> {
+    impl Trgs {
       #[inline(always)]
       fn get_trg(&self, i: TrgId) -> Trg {
-        unsafe { (*self.0.as_ptr().byte_offset(i.byte_offset as _)).clone() }
+        unsafe { (*self.0.byte_offset(i.byte_offset as _)).clone() }
       }
 
       #[inline(always)]
       fn set_trg(&mut self, i: TrgId, trg: Trg) {
-        unsafe { *self.0.as_mut_ptr().byte_offset(i.byte_offset as _) = trg }
+        unsafe { *self.0.byte_offset(i.byte_offset as _) = trg }
       }
     }
```

Comment:
The reason why this doesn't buffer overflow (and `Trgs`'s invariants) should be described here.